components/scream/src/share/property_checks/field_within_interval_check.cpp

```diff
@@ -114,13 +170,65 @@ PropertyCheck::CheckResult FieldWithinIntervalCheck::check_impl () const
           "You should not have reached this line. Please, contact developers.\n");
   }
   PropertyCheck::CheckResult check_result;
-  check_result.pass = minmax.min_val>=m_lower_bound && minmax.max_val<=m_upper_bound;
-  check_result.msg = "";
+  check_result.pass = minmaxloc.min_val>=m_lower_bound && minmaxloc.max_val<=m_upper_bound;
   if (not check_result.pass) {
-    check_result.msg = std::string("FieldWithinIntervalCheck failed")
-                     + "; min = " + std::to_string(minmax.min_val)
-                     + "; max = " + std::to_string(minmax.max_val) + "\n";
+    check_result.msg  = "Check failed.\n";
+    check_result.msg += "  " + this->name() + "\n";
+  } else {
+    check_result.msg  = "Check passed.\n";
+    check_result.msg += "  " + this->name() + "\n";
+  }
+
+  auto idx_min = unflatten_idx(layout.dims(),minmaxloc.min_loc);
+  auto idx_max = unflatten_idx(layout.dims(),minmaxloc.max_loc);
+
+  using namespace ShortFieldTagsNames;
+
+  int min_col_lid, max_col_lid;
+  AbstractGrid::dofs_list_h_type gids;
+  AbstractGrid::geo_view_h_type lat, lon;
+  bool has_latlon;
+  bool has_col_info = m_grid and layout.tag(0)==COL;
+
+  if (has_col_info) {
+    // We are storing grid info, and the field is over columns. Get col id and coords.
+    min_col_lid = idx_min[0];
+    max_col_lid = idx_max[0];
+    gids = m_grid->get_dofs_gids_host();
+    has_latlon = m_grid->has_geometry_data("lat") && m_grid->has_geometry_data("lon");
+    if (has_latlon) {
+      lat = m_grid->get_geometry_data_host("lat");
+      lon = m_grid->get_geometry_data_host("lon");
+    }
+  }
+
+
+  check_result.msg += "  minimum:\n";
+  check_result.msg += "    - value: " + std::to_string(minmaxloc.min_val) + "\n";
```

Comment:
@bartgol - One thing I'm noticing today is that the min and max values are currently getting printed as floats instead of in exponential notation. This makes it hard to tell how bad roundoff errors are: we're getting messages like "x is outside [0,1]. Min=0.0000 and max=1.0000". Would it be easy to use exponential notation for just the min and max (and not the lat/lon)?

Reply:
Ah, yeah, good point. I can take care of that.